src/helm/proxy/clients/client.py

```diff
@@ -79,14 +77,10 @@ def make_cache_key(raw_request: Dict, request: Request) -> Dict:
         return cache_key
 
     def tokenize(self, request: TokenizationRequest) -> TokenizationRequestResult:
-        # Deprecated - use `self.tokenizer.tokenize` instead. Warn the user.
-        hlog("WARNING: CachingClient.tokenize is deprecated, use self.tokenizer.tokenize instead")
-        return self.tokenizer.tokenize(request)
+        raise NotImplementedError("CachingClient.tokenize() is not supported, use self.tokenizer.tokenize instead")
```

Comment:
This comment should be conditioned on the fact that the model has a tokenizer or not. I personally liked the idea of having a `Tokenizer` in `Client`, maybe just make it `Optional` and check here that it is not None?

Reply:
This makes mypy unhappy because then you would need to `assert self.tokenizer` before you call `self.tokenizer.tokenize()` in multiple files (`palmyra_client`, `openai_client`, `megatron_client`, `anthropic_client`, etc.) because now `self.tokenizer` can be `None` in those files.

I intend to delete this method completely in a subsequent PR, anyway.

Reply:
These methods have now been deleted thanks to #1994